apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -97,7 +97,14 @@ def __init__(
         self.array = array
         self.sr = sr
         self.eager = eager
-        self.eager_iterator_pool = eager_iterator_pool
+
+        # Assign a thread pool from the context, or create a new one if no context
+        # is available
+        if context is not None:
+            self._threadpool = context.threadpool
+        else:
+            self._threadpool = futures.ThreadPoolExecutor()
```

Comment:
since `context` is (presumably) always going to be a truthy object, this can be `if context` (thus reducing the amount of negatives)